src/strategy/HsiStrategy.py

```diff
@@ -56,6 +57,15 @@ def algo_container():
     algo_result = get_algo_result(asset_price, weights, "CRP")
     algo_result.fee = 0.2 / 100
     print(algo_result.summary())
+
+    equity_curve_df = pd.DataFrame(
+        {algo_result.algo_name: algo_result.equity_curve},
+    )
+
```

Comment:
Please use the code in 80-86. The type of index result is algo_result.
equity_curve_df = pd.DataFrame({
  algo_result.algo_name: algo_result.equity_curve,
  index_result.algo_name:  index_result.equity_curve
})